astropy/units/format/generic.py

```diff
@@ -464,11 +466,37 @@ def _parse_unit(cls, s, detailed_exception=True):
         else:
             raise ValueError()
 
+    _translations = str.maketrans({
+        '\N{GREEK SMALL LETTER MU}': '\N{MICRO SIGN}',
+        '\N{MINUS SIGN}': '-',
+        '\N{SUPERSCRIPT MINUS}': '-',
+        '\N{SUPERSCRIPT PLUS SIGN}': '+',
+        '\N{SUPERSCRIPT ZERO}': '0',
```

Comment:
I like this but it also worries me a bit - as I think one should check that there is not a mix of superscript and non-superscript numbers. So, I *think* I'd rather do this in `p_numeric_power` - allow new tokens `superscript_sign superscript_uint`  - or pre-translate but check that there are no regular numbers next to it.

Reply:
merged your PR